cocotb/build_libs.py

```diff
@@ -0,0 +1,420 @@
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import sys
+import sysconfig
+import logging
+import distutils
+
+from setuptools import Extension
+from distutils.spawn import find_executable
+from setuptools.command.build_ext import build_ext as _build_ext
+from distutils.file_util import copy_file
+
+
+logger = logging.getLogger(__name__)
+
+
+class build_ext(_build_ext):
+
+    # Needed for Windows to not assume python module (generate interface in def file)
+    def get_export_symbols(self, ext):
+        return None
+
+    # For proper cocotb library nameing, based on https://github.com/cython/cython/issues/1740
+    # TODO: for the simulator python library, we should be leaving this suffix in place
+    def get_ext_filename(self, ext_name):
+        """
+        Like the base class method, but removes the ``.cpython-36m-x86_64-linux-gnu.`` part before the extension.
+
+        Also replaces ``.pyd`` with ``.dll`` on windows.
+        """
+
+        filename = _build_ext.get_ext_filename(self, ext_name)
+
+        head, tail = os.path.split(filename)
+        tail_split = tail.split(".")
+
+        if os.name == "nt":
+            ext_name = "dll"
+        else:
+            ext_name = "so"
+
+        filename_short = os.path.join(head, tail_split[0] + "." + ext_name)
+
+        # icarus requires vpl extension
+        if filename.find("icarus") >= 0:
+            filename_short = filename_short.replace("libvpi.so", "gpivpi.vpl")
+            filename_short = filename_short.replace("libvpi.dll", "gpivpi.vpl")
```

Comment:
Why the rename from `lib` to `gpi`? Would be nice to indicate the reason in the comment too.

Reply:
This is how it is now in the makefiles. I propose to make new PR when we use unique names for all simulators.